client/post-editor/media-modal/index.jsx

```diff
@@ -253,7 +255,10 @@ export class EditorMediaModal extends Component {
 				fileName: fileName,
 				fileContents: blob,
 				mimeType: mimeType
-			}
+			},
+			// conditionally add the height and width of the transient image
+			...width && { width },
+			...height && { height }
```

Comment:
While this is a pretty high answer on stackoverflow, and supposedly allowed by the [spec](https://github.com/tc39/proposal-object-rest-spread/issues/45) using a rest operator on a non-iterable item feels very clever (we even have a comment!), and may confuse folks who are not familiar with the language. Unless folks feel very strongly about this, I think a plain Object.assign would be clearer here.